src/query/functions/temporal/rate.go

```diff
@@ -49,42 +49,53 @@ const (
 	IncreaseType = "increase"
 )
 
-type rateProcessor struct {
-	isRate, isCounter bool
-	rateFn            rateFn
+// RateProcessor is a structure containing details about the rate.
+type RateProcessor struct {
```

Comment:
Does this need to be exported?

Reply:
Believe we want to be able to create custom rate processors yes.

@arnikola is it worth creating a new test file so that we ensure we can create custom ones? i.e. someone doesn't accidently unexport without a test failing?

i.e.
`rate_export_test.go`
```go
// license header
package temporal_test // this is allowed in same package if it's a test file

import (
  "testing"

  "github.com/m3db/m3/src/query/functions/temporal" // need to import since this is in "temporal_test" package even though in same package directory
)

func TestRateProcessorCustomRateFn(t *testing.T) {
  p := &temporal.RateProcessor{ // this will look like a real import
    RateFn: func() { ... },
  }
  ... = p.process(...)
  // verify minimal thing working
}
```

Reply:
Yeah sorry working on a test for rate functions that requires this to be exported, but may leave the test itself as a todo for the immediate future